### PR TITLE
completed warning modal and is testable at /donor-form

### PR DESCRIPTION
**Pull request by mollyxu**
Warning modal that appears when user accesses the donor form on mobile devices.
Can be tested at /donor-form

image

### CODE REVIEW

File: client/src/components/layouts/DonationFormPage.tsx [hunk 1]
```diff
@@ -22,60 +25,87 @@ interface Props {
 }
 
 const DonationFormPage: FunctionComponent<Props> = (props: Props) => {
+    const [show, setShow] = useState(true);
+
+    const handleClose = () => setShow(false);
+    // const handleShow = () => setShow(true);
```
Comment (meganniu) [inline client/src/components/layouts/DonationFormPage.tsx]: remove comment

File: client/src/components/layouts/DonationFormPage.tsx [hunk 2]
```diff
@@ -22,60 +25,87 @@ interface Props {
...
+    React.useEffect(() => {
+      const handleWindowResize = () => setWidth(window.innerWidth)
+      window.addEventListener("resize", handleWindowResize);
+      // Return a function from the effect that removes the event listener
+    //   return () => window.removeEventListener("resize", handleWindowResize);
+      width < breakpoint ? setShow(true):setShow(false)
+      
+    }, []);
+
+
```
Comment (meganniu) [inline client/src/components/layouts/DonationFormPage.tsx]: nit: remove two newlines

File: client/src/components/layouts/DonationFormPage.tsx [hunk 3]
```diff
@@ -22,60 +25,87 @@ interface Props {
 }
 
 const DonationFormPage: FunctionComponent<Props> = (props: Props) => {
+    const [show, setShow] = useState(true);
```
Comment (meganniu) [inline client/src/components/layouts/DonationFormPage.tsx]: nit: use more specific name like `showMobilePopup`

File: client/src/components/layouts/DonationFormPage.tsx [hunk 4]
```diff
@@ -22,60 +25,87 @@ interface Props {
...
+
+    const [width, setWidth] = React.useState(window.innerWidth);
+    const breakpoint = 576;
+  
+    React.useEffect(() => {
+      const handleWindowResize = () => setWidth(window.innerWidth)
+      window.addEventListener("resize", handleWindowResize);
+      // Return a function from the effect that removes the event listener
+    //   return () => window.removeEventListener("resize", handleWindowResize);
+      width < breakpoint ? setShow(true):setShow(false)
```
Comment (meganniu) [inline client/src/components/layouts/DonationFormPage.tsx]: nit: use this instead:
```
setShow(width < breakpoint)
```

File: client/src/components/layouts/DonationFormPage.tsx [hunk 5]
```diff
@@ -22,60 +25,87 @@ interface Props {
...
+    // const handleShow = () => setShow(true);
+
+    const [width, setWidth] = React.useState(window.innerWidth);
+    const breakpoint = 576;
+  
+    React.useEffect(() => {
+      const handleWindowResize = () => setWidth(window.innerWidth)
+      window.addEventListener("resize", handleWindowResize);
+      // Return a function from the effect that removes the event listener
+    //   return () => window.removeEventListener("resize", handleWindowResize);
```
Comment (meganniu) [inline client/src/components/layouts/DonationFormPage.tsx]: nit: remove comment

File: client/src/components/layouts/DonationFormPage.tsx [hunk 6]
```diff
@@ -22,60 +25,87 @@ interface Props {
 }
 
 const DonationFormPage: FunctionComponent<Props> = (props: Props) => {
+    const [show, setShow] = useState(true);
+
+    const handleClose = () => setShow(false);
+    // const handleShow = () => setShow(true);
+
+    const [width, setWidth] = React.useState(window.innerWidth);
```
Comment (meganniu) [inline client/src/components/layouts/DonationFormPage.tsx]: nit: use more specific name like `setMobilePopupWidth`

File: client/src/components/layouts/DonationFormPage.tsx [hunk 7]
```diff
@@ -1,10 +1,13 @@
-import React, { FunctionComponent } from "react";
+import React, { FunctionComponent, useState } from "react";
 
 import { Button } from "../atoms/Button";
 import Navbar from "../organisms/Navbar";
 import StepNumber from "../atoms/StepNumber";
 import Stepper from "../atoms/Stepper";
```
Comment (meganniu) [inline client/src/components/layouts/DonationFormPage.tsx]: nit: remove newline

**Review by meganniu — APPROVED** [review]
Nice work @mollyxu !!
Thanks for getting this done so quickly :))

You'll need to run `npm run pretty` on `client` before pushing

**Review by kevin-wangg — COMMENTED** [review]
LGTM! Thanks Molly!

**Review by kevin-wangg — APPROVED** [review]
LGTM!

**Pull request merged**